modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java

```diff
@@ -122,9 +118,21 @@ public void onPistonEvent(BlockPistonExtendEvent event) {
         Block block = event.getBlock();
         CraftManager.getInstance().getCraftsInWorld(block.getWorld());
         for (Craft tcraft : CraftManager.getInstance().getCraftsInWorld(block.getWorld())) {
-            MovecraftLocation mloc = new MovecraftLocation(block.getX(), block.getY(), block.getZ());
-            if (MathUtils.locIsNearCraftFast(tcraft, mloc) && tcraft.getCruising() && !tcraft.isNotProcessing()) {
-                event.setCancelled(true);
+            if(tcraft == null) continue;
+
+            if (MathUtils.locationInHitBox(tcraft.getHitBox(), block.getLocation())) {
+                if (tcraft.getCruising() && !tcraft.isNotProcessing()) {
+                    event.setCancelled(true);
+                }
+                else if(tcraft.getType().getBoolProperty(CraftType.MERGE_PISTON_EXTENSIONS)){
+                    SetHitBox hitBox = new SetHitBox();
```

Comment:
Is there a reason you used a SetHitBox?

Reply:
I actually can't remember why now. Would it be better to use a BitmapHitbox?

Reply:
It's our most performant hitbox currently.